File: uploader-clean/src/main/java/net/hearthstats/util/AnalyticsTracker.java
package net.hearthstats.util;

import com.dmurph.tracking.AnalyticsConfigData;
import com.dmurph.tracking.JGoogleAnalyticsTracker;
import com.dmurph.tracking.JGoogleAnalyticsTracker.GoogleAnalyticsVersion;

/**
 * Factory for Google Analytics Tracker
 * 
 * @author tyrcho
 */
public final class AnalyticsTracker {
	// Singleton
	private AnalyticsTracker() {
	}

	public static JGoogleAnalyticsTracker tracker() {
		return new JGoogleAnalyticsTracker(new AnalyticsConfigData(
				"UA-45442103-3"), GoogleAnalyticsVersion.V_4_7_2);
	}
}

File: companion/src/main/scala/net/hearthstats/util/AnalyticsTracker.scala
package net.hearthstats.util

import com.dmurph.tracking.JGoogleAnalyticsTracker
import com.dmurph.tracking.AnalyticsConfigData
import com.dmurph.tracking.JGoogleAnalyticsTracker.GoogleAnalyticsVersion._

object AnalyticsTrackerFactory {
  def tracker(enabled: Boolean): Tracker =
    if (enabled) GoogleAnalyticsTracker
    else NoTracker
}

sealed trait Tracker {
  def trackEvent(category: String, action: String): Unit
}

object NoTracker extends Tracker {
  def trackEvent(category: String, action: String): Unit = {}
}

object GoogleAnalyticsTracker extends Tracker {
  val impl = new JGoogleAnalyticsTracker(new AnalyticsConfigData("UA-45442103-3"), V_4_7_2)

  def trackEvent(category: String, action: String): Unit = {
    impl.trackEvent(category, action)
  }
}